flycheck.el

```diff
@@ -7744,7 +7744,12 @@ Otherwise return the previously used cache directory."
 Look up the directory hierarchy from FILE for a directory
 containing a file that matches REGEXP."
   (locate-dominating-file
-   file
+   ;; We must not pass regular file here because
+   ;; `locate-dominating-file' will just pass it into its second
+   ;; lambda argument . But our lambda expects only directories.
+   (if (file-directory-p file)
+       file
+     (file-name-directory file))
```

Comment:
> But if we actually have been using locate-dominating-file wrong all this time, then there is no reason to held onto it. (Note: we should probably hunt these down in the code now)

Well, we can find solace in knowing that we're note alone :) https://debbugs.gnu.org/cgi/bugreport.cgi?bug=27798#11

@sergv Thanks for your efforts and patience